chain/transaction.go

```diff
@@ -185,56 +171,56 @@ func (tx *Transaction) RLPHash() util.Hash {
 	return util.DoHash(bytes)
 }
 
-// EncodeRLP writes p as RLP list [a, b] that omits the Name field.
-func (tx *Transaction) EncodeRLP(w io.Writer) (err error) {
-	if tx == nil {
-		err = rlp.Encode(w, []uint{0, 0, 0, 0})
-	} else {
-		var blkNum0 uint64
-		var txIdx0 uint32
-		var outIdx0 uint8
-		var blkNum1 uint64
-		var txIdx1 uint32
-		var outIdx1 uint8
-		var newOwner0 common.Address
-		var amount0 *big.Int
-		var newOwner1 common.Address
-		var amount1 *big.Int
-
-		if tx.Input0 != nil {
-			blkNum0 = tx.Input0.BlkNum
-			txIdx0 = tx.Input0.TxIdx
-			outIdx0 = tx.Input0.OutIdx
-		}
-
-		if tx.Input1 != nil {
-			blkNum1 = tx.Input1.BlkNum
-			txIdx1 = tx.Input1.TxIdx
-			outIdx1 = tx.Input1.OutIdx
-		}
-
-		if tx.Output0 != nil {
-			newOwner0 = tx.Output0.NewOwner
-			amount0 = tx.Output0.Amount
-		}
-
-		if tx.Output1 != nil {
-			newOwner1 = tx.Output1.NewOwner
-			amount1 = tx.Output1.Amount
-		}
-
-		err = rlp.Encode(w, []interface{}{
-			blkNum0,
-			txIdx0,
-			outIdx0,
-			blkNum1,
-			txIdx1,
-			outIdx1,
-			newOwner0,
-			amount0,
-			newOwner1,
-			amount1,
-		})
-	}
-	return err
+func (tx *Transaction) EncodeRLP(w io.Writer) error {
+	var itf rlpHelper
+	if tx.Input0 != nil {
+		itf.BlkNum0 = tx.Input0.BlkNum
+		itf.TxIdx0  = tx.Input0.TxIdx
+		itf.OutIdx0 = tx.Input0.OutIdx
+		itf.Sig0    = tx.Sig0
+	}
+	if tx.Input1 != nil {
+		itf.BlkNum1 = tx.Input1.BlkNum
+		itf.TxIdx1  = tx.Input1.TxIdx
+		itf.OutIdx1 = tx.Input1.OutIdx
+		itf.Sig1    = tx.Sig1
+	}
+	if tx.Output0 != nil {
+		itf.NewOwner0 = tx.Output0.NewOwner
+		itf.Amount0   = *tx.Output0.Amount
+	}
+	if tx.Output1 != nil {
+		itf.NewOwner1 = tx.Output1.NewOwner
+		itf.Amount1   = *tx.Output1.Amount
+	}
+	if tx.Fee != nil {
+		itf.Fee = *tx.Fee
+	}
+	return rlp.Encode(w, &itf)
 }
+
+func (tx *Transaction) DecodeRLP(s *rlp.Stream) error {
+	var itf rlpHelper
+	err := s.Decode(&itf)
+	if err != nil {
+		return err
+	}
+	tx.Input0  = ZeroInput()
```

Comment:
Could leverage constructor pattern for these:

`tx.input0 = NewInput(blkNum0, txIdx0)`

Reply:
Fixed.